back/engines/commercial/id_vienna_saml/spec/requests/shared.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.shared_context 'with Vienna SAML authentication enabled' do |provider_name|
+  define_method :enable_vienna_login do
```

Comment:
I guess `def enable_vienna_login` was not working, maybe `def self.enable_vienna_login`? I try to avoid using `define_method` when I can.

Reply:
I need to access the `provider_name` block parameter in these methods, and the Ruby method definitions `def` or `def self.` do not allow it.